### ISSUE TITLE
feat: enable colours for Parties

### ISSUE THREAD

**Issue by kaljarv** [OpenVAA/voting-advice-application#374]
Categories need to have a theme colour,  this can be added as a `color` property of the `Question Category` objects. `Parties` already have this, but it has no effect in the frontend.

The color should be applied to the category term when shown next to the questions and possibly in category headings (in the Candidate App). This implies changes to the frontend data types, such that the `category` of a `QuestionProp` is an object instead of just a string containing the name.

For parties, the colour should be applied to the party flag icon. This, in turn, would possibly imply changes to the `Icon` component to allow for a custom color in addition to the named ones. This would differ from the so that the it would:
- either automatically create a different version for use in dark mode
- accept to colors, `customColorLight` and `customColorDark` for use in light and dark modes

Finally, before using any of the colors in the frontend, the colour should be automatically modified to have sufficient contrast on `base-1/300`. Possibly:
- use the formula for contrast calculation in this [discussion](https://stackoverflow.com/questions/65304462/is-it-possible-to-make-a-javascript-program-that-changes-the-color-of-the-text-t)
- either lighten or darken by mixing with black or white and either solve for correct color analytically or iteratively ([example here](https://medium.com/confrere/automatically-creating-an-accessible-color-palette-from-any-color-sure-e735c3f2f45e), although it uses a dep)
